### ISSUE TITLE
关于噪声处理的问题？

### ISSUE THREAD

**Issue by Nonbiuld** [HanxunH/Unlearnable-Examples#13]
训练生成的噪声的数值是有正有负的，下面这个操作把负的值都变成0了，这种做法不会有问题吗？不应该是噪声加图像再clamp到0-255？
self.perturb_tensor.mul(255).clamp_(0, 255).permute(0, 2, 3, 1).to('cpu').numpy()

**Comment by HanxunH** [comment]
Yes, the noise should be added first. This could improve the protection performance. Keep me updated if you notice any difference in the results.

**Comment by Nonbiuld** [comment]
训练生成噪声的代码似乎也有一点小问题，images[i] = images[i] + sample_noise，比如这里这个image加了sample_noise之后，是不是需要再clamp到(0,1)会更加合理一点，如果更改后效果有所提升我会告诉你的。

**Comment by HanxunH** [comment]
Good catch, thanks!

**Comment by pzs19** [comment]
@Nonbiuld 老哥，我也注意到这个问题了，请问更改之后效果如何？
